docs/contributing.md

```diff
@@ -0,0 +1,108 @@
+
+* Help improve the Kusion documentation
+* Clarify code, variables, or functions that can be renamed or commented on
+* Write test coverage
+* Help triage issues
+
+After finding an interested task, you need to [open an issue](https://github.com/KusionStack/kusion/issues) to track it. Now, Kusion provides three issue templates as follows, please choose one according to your need:
+
+* Bug Report: Report a bug encountered while operating Kusion
+* Enhancement Tracking Issue: Provide supporting details for a feature in development
+* Failing Test: Report continuously failing tests or jobs in Kusion CI
+
+In the issue, please describe your problem clearly and accurately, and choose a label to define the issue's type. There are the commonly used label:
+
+* area/cli: Categorizes an issue or PR as relevant to CLI
+* area/engine: Categorizes an issue or PR as relevant to Kusion engine
+* area/release: Categorizes an issue or PR as relevant to build
+* area/runtime: All Infrastructure Runtime, eg: k8s
+* area/testing: Categorizes an issue or PR as relevant to test
+* kind/feature: Categorizes issue or PR as related to a new feature
+
+## Contribute a Pull Request
+
+After opening an issue, you could contribute codes to Kusion by a pull request. Here are the steps you should follow:
+
+### Fork Repository
+
+Kusion adopts trunk-based development, i.e., the code used for release is maintained on the main branch. And to keep the repository tidy, Kusion only has the main branch. 
+
+Thus, to develop Kusion, you have to fork the [KusionStack/kusion](https://github.com/KusionStack/kusion) repository to your own workspace, then checkout a new branch to develop coding.
+
+### Develop Code
+
+Now you can start coding to solve the issue. To maintain the code quality of Kusion, unit tests are indispensable for the functions you added or updated.
+
+After the development is completed, commit and push to your own forked repository. Cause the code will merge into Kusion repository by pull request, [commitlint/config-conventional](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) should get followed, the legal prefixes for commit message are shown below:
+
+* build
+* chore
+* ci 
+* docs 
+* feat 
+* fix 
+* perf 
+* refactor 
+* revert 
+* style 
+* test
+
+### Open a Pull Request
+
+[Open a pull request](https://github.com/KusionStack/kusion/pulls) from the develop branch of your forked repository to the main branch of Kusion. You should clearly describe what you do in the PR, and link it to an issue. Besides, the PR title should also follow the commit conventions described above, and must be 5-80 characters in length.
+
+### Sign CLA
+
+If it was your first pull request, you need to sign our [CLA(Contributor License Agreement)](https://github.com/KusionStack/.github/blob/main/CLA.md). The only thing you need to do is to post a pull request comment same as the below format:
+
+---
```

Comment:
The horizontal divider line may cause users to be confused about whether to copy this line or not.  Setting this message as a copiable code style will be helpful.

Reply:
The horizontal divider line style is the same as what is IN REAL shown in the pull request comment

Reply:
I know that. The default style is misleading and some new contributors have problems with this. e.g. https://github.com/KusionStack/konfig/pull/119

I have created an issue to track this. https://github.com/KusionStack/kusion/issues/365